src/main/java/io/quarkus/search/app/hibernate/QuarkusIOLoadingContext.java

```diff
@@ -31,6 +34,13 @@ public List<Guide> nextBatch(int batchSize) {
         for (int i = 0; guides.hasNext() && i < batchSize; i++) {
             list.add(guides.next());
         }
+        if (!guides.hasNext()) {
+            // This guide stream is a concatenated, where one of the streams is "file-based" and such stream requires to be closed.
+            // Once there are no more guides in the stream:
+            // we are closing the concatenated stream which will in turn propagate this to streams it is created from.
+            // Which eventually will result in file-stream releasing its resources.
+            guideStream.close();
+        }
```

Comment:
I guess... this should do the trick. Previously, we were creating an iterator from the stream and never called `close()`.